src/librustdoc/html/static/main.js

```diff
@@ -1422,11 +1424,13 @@ function hideThemeButtonState() {
             // errors in mobile browsers).
             if (e.tagName === "H2" || e.tagName === "H3") {
                 var nextTagName = e.nextElementSibling.tagName;
-                if (nextTagName == "H2" || nextTagName == "H3") {
+                if (nextTagName === "H2" || nextTagName === "H3") {
                     e.nextElementSibling.style.display = "flex";
-                } else {
+                } else if (nextTagName !== "DETAILS") {
                     e.nextElementSibling.style.display = "block";
                 }
+            } else if (e.tagName === "DETAILS") {
+                e.open = true;
```

Comment:
Not all details should be opened by details. For example the traits implementations blocks.

Reply:
Or did you mean that we should generate the `<details>` with "open" directly? Still: we will need this JS in any case for when we press the global toggle.

Reply:
Yes, I mean that for default-shown sections we should generate `<details open>`.

I agree we need JS to make `<details>` respond to the global toggle, but I believe that's already in place. And it wouldn't be in this chunk of code, which executes on load rather than in response to the global toggle.

Reply:
You're right, I'll modify it then.